### PR TITLE
Markup: auto-link plural forms of terms

### PR DESCRIPTION
**Pull request by bakkot**
I [just added](https://github.com/tc39/ecmarkup/pull/346) support in ecmarkup for `dfn` tags to define other forms of the terms they define. This PR bumps ecmarkup to that version and makes use of the new attribute as appropriate for plural forms of defined nouns and noun phrases.

### CODE REVIEW

**Comment by ljharb** [comment]
Lots of boilerplate from not auto-handling the "s" form :-/

**Comment by bakkot** [comment]
Less than you might think - there's a lot of `dfn`s which aren't nouns and which therefore should not have `-s` forms, so we'd need at least a marker to indicate that a particular term is a noun and should be pluralized. Given that, also including the plural isn't much additional overhead.

File: spec.html [hunk 1]
```diff
@@ -2406,14 +2406,14 @@ <h1>The Object Type</h1>
...
-      <p>An <dfn id="integer-index">integer index</dfn> is a String-valued property key that is a canonical numeric String (see <emu-xref href="#sec-canonicalnumericindexstring"></emu-xref>) and whose numeric value is either *+0*<sub>𝔽</sub> or a positive integral Number &le; 𝔽(2<sup>53</sup> - 1). An <dfn id="array-index">array index</dfn> is an integer index whose numeric value _i_ is in the range <emu-eqn>*+0*<sub>𝔽</sub> &le; _i_ &lt; 𝔽(2<sup>32</sup> - 1)</emu-eqn>.</p>
+      <p>An <dfn id="integer-index" variants="integer indexes">integer index</dfn> is a String-valued property key that is a canonical numeric String (see <emu-xref href="#sec-canonicalnumericindexstring"></emu-xref>) and whose numeric value is either *+0*<sub>𝔽</sub> or a positive integral Number &le; 𝔽(2<sup>53</sup> - 1). An <dfn id="array-index" variants="array indexes">array index</dfn> is an integer index whose numeric value _i_ is in the range <emu-eqn>*+0*<sub>𝔽</sub> &le; _i_ &lt; 𝔽(2<sup>32</sup> - 1)</emu-eqn>.</p>
```
Comment (michaelficarra) [inline spec.html:2416]: We should settle on one of "indexes" or "indices". There's 2 occurrences of "indexes" (1 of each of these plurals) currently and 11 of "indices".

Comment (gibson042) [reply spec.html:2416]: [ECMA-402](https://tc39.es/ecma402/) also has a single occurrence of "indices" and none of "indexes", and the [Segmenter proposal](https://tc39.es/proposal-intl-segmenter/) has two of "indices" and none of "indexes".

Comment (bakkot) [reply spec.html:2416]: I'm happy to change to be consistently "indices" everywhere.

File: spec.html [hunk 2]
```diff
@@ -13662,7 +13662,7 @@ <h1>Array Exotic Objects</h1>
         <p>A String property name _P_ is an <em>array index</em> if and only if ToString(ToUint32(_P_)) equals _P_ and ToUint32(_P_) is not the same value as 𝔽(<emu-eqn>2<sup>32</sup> - 1</emu-eqn>).</p>
       </emu-note>
 
-      <p>An object is an <dfn id="array-exotic-object">Array exotic object</dfn> (or simply, an Array object) if its [[DefineOwnProperty]] internal method uses the following implementation, and its other essential internal methods use the definitions found in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots"></emu-xref>. These methods are installed in ArrayCreate.</p>
+      <p>An object is an <dfn id="array-exotic-object" variants="Array exotic object">Array exotic object</dfn> (or simply, an Array object) if its [[DefineOwnProperty]] internal method uses the following implementation, and its other essential internal methods use the definitions found in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots"></emu-xref>. These methods are installed in ArrayCreate.</p>
```
Comment (michaelficarra) [inline spec.html:13665]:
```suggestion
      <p>An object is an <dfn id="array-exotic-object" variants="Array exotic objects">Array exotic object</dfn> (or simply, an Array object) if its [[DefineOwnProperty]] internal method uses the following implementation, and its other essential internal methods use the definitions found in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots"></emu-xref>. These methods are installed in ArrayCreate.</p>
```

File: spec.html [hunk 3]
```diff
@@ -42019,7 +42019,7 @@ <h1>FinalizationRegistry Objects</h1>
 
     <emu-clause id="sec-finalization-registry-constructor">
       <h1>The FinalizationRegistry Constructor</h1>
-      <p>The <dfn>FinalizationRegistry</dfn> constructor:</p>
+      <p>The <dfn variants="FinalizationRegistries">FinalizationRegistry</dfn> constructor:</p>
```
Comment (michaelficarra) [inline spec.html:42022]: I'm still kind of partial to `FinalizationRegistrys`.

File: spec.html (hunk 2 above)
Comment (michaelficarra) [reply spec.html:13665]: This is a double dfn. Shouldn't the linter complain?

**Review by michaelficarra — APPROVED** [review]
LGTM otherwise. I didn't review for any new unintended links, but I think we can just address those as we find them.

File: spec.html (hunk 1 above)
Comment (michaelficarra) [reply spec.html:2416]: Let's just address it here. I don't think it's worth another issue/PR.

File: spec.html (hunk 2 above)
Comment (bakkot) [reply spec.html:13665]: There is not currently a lint rule for definitions which occur multiple times, though yeah there probably should be.

**Pull request merged**